ouroboros-network/src/Ouroboros/Network/PeerSelection/RootPeersDNS.hs

```diff
@@ -9,8 +9,23 @@
 --  'resolverResource' and 'asyncResolverResource' are not used when compiled
 --  on @Windows@
 {-# OPTIONS_GHC -Wno-unused-top-binds #-}
+{-# OPTIONS_GHC -Wno-unused-matches #-}
 
 module Ouroboros.Network.PeerSelection.RootPeersDNS (
+    -- * DNS based actions for local and public root providers
+    DNSActions (..),
+
+    -- * DNS resolver IO auxiliar functions
+    constantResource,
+    resolverResource,
+    asyncResolverResource,
+#if defined(mingw32_HOST_OS)
+    newResolverResource,
+#endif
+    lookupAWithTTL,
```

Comment:
Do we need to export all of them; some of them are already exported as `ioDNSActions`.

Reply:
`DNSActions` is more general than what `RootPeersDNS` provides, I think it makes sense to add `Ouroboros.Network.DNSActions` module which provides the type, `ioDNSActions` and possibly auxiliary functions.

Reply:
`resolveDomainAddresses` could also be moved to `DNSActions` module.

Reply:
Okay, that is a good idea! Will do!